js/index.js

```diff
@@ -0,0 +1,325 @@
+                return true;
+            }
+            else {
+                swal("Enter unique Product Name!", {
+                    icon: "warning",
+                    timer: 2000
+                });
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+//Function to clear data after using modal
+function clearValue() {
+    $('#addAdmin').modal('hide');
+    document.getElementById("productid").value = "";
+    document.getElementById("productname").value = "";
+    document.getElementById("image").value = "";
+    document.getElementById("imgedit").src = "";
+    document.getElementById("price").value = "";
+    document.getElementById("description").value = "";
+}
+
+//Function to sort/filter
+function changeBy(x) {
+    var productList;
+    productList = (localStorage.getItem("productList") == null) ? productList = [] : JSON.parse(localStorage.getItem("productList"));
+    switch (x) {
+        case "lowFirst":
+            productList.sort((a, b) => {
+                return a.price - b.price;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+        case "highFirst":
+            productList.sort((a, b) => {
+                return b.price - a.price;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+        case "idFirst":
+            productList.sort((a, b) => {
+                return a.productid - b.productid;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+        case "nameFirst":
+            productList.sort(function (a, b) {
+                var x = a.productname.toLowerCase();
+                var y = b.productname.toLowerCase();
+                return x < y ? -1 : x > y ? 1 : 0;
+            });
+            localStorage.setItem('productList', JSON.stringify(productList));
+            showData();
+            break;
+    }
+}
+
+//Function to show image while adding or editing
+function readURL(input) {
+    document.getElementById("image").src = "";
+    if (input.files || input.files[0]) {
+        var reader = new FileReader();
+        reader.onload = function (e) {
+            $('#imgedit')
+                .attr('src', e.target.result)
+                .width(80)
+                .height(80);
+        };
+
+        reader.readAsDataURL(input.files[0]);
+    }
+}
+
+//Function to show the input data 
+function showData() {
+    var productList;
+    if (localStorage.getItem("productList") == null) {
+        productList = [];
+    }
+    else {
+        productList = JSON.parse(localStorage.getItem("productList"));
+        var html = "";
+        productList.forEach(function (element, index) {
+            html += "<tr>";
```

Comment:
Is it possible we can use here string interpolation instead of concatenation string? 
i.e
https://www.geeksforgeeks.org/string-interpolation-in-javascript/

Reply:
solved